credentials/alts/internal/testutil/testutil.go

```diff
@@ -123,3 +126,154 @@ func MakeFrame(pl string) []byte {
 	copy(f[conn.MsgLenFieldSize:], []byte(pl))
 	return f
 }
+
+// FakeHandshaker is a fake implementation of the ALTS handshaker service.
+type FakeHandshaker struct {
+	altspb.HandshakerServiceServer
+}
+
+// DoHandshake performs a fake ALTS handshake.
+func (h *FakeHandshaker) DoHandshake(stream altspb.HandshakerService_DoHandshakeServer) error {
```

Comment:
Frankly, it is a lot of code for a fake server? Is it at all possible to use a real server instead, and give the tests real credentials to go with. We use real TLS certs for most of our tests. I don't know how possible/feasible it is for ALTS though.

Reply:
Ya I agree, it's an unfortunate situation. For some context: all of the tests that use real ALTS credentials are run in google3 once the code is imported, and (before this PR) there are no e2e-style tests for ALTS in the OSS repo. This was done for 2 reasons:

1. ALTS credentials are proprietary and we do not want them to be checked into an OSS repo.
2. Because of security policies, we are not allowed to generate "test ALTS private keys" and check them into an OSS repo.

Because of these restrictions, if we want to have some sort of e2e test for ALTS in the OSS repo (like is added in this PR), then we need to completely fake out the handshaker service. Totally agreed that this is not perfect, but it's the same situation as what is done in C-core (and it's on my TODO list to add this to Java as well).

Reply:
Ack.